### PR TITLE
drivers: counter: Add Ambiq counter driver

### PR DESCRIPTION
**Pull request by msierszulski**
This PR adds the Ambiq counter driver, as well as instances in the SoC dtsi file, enabling it on the apollo4p_evb board.

It should be merged after https://github.com/zephyrproject-rtos/hal_ambiq/pull/2 is merged.

### CODE REVIEW

**Review by fkokosinski — APPROVED** [review]
LGTM

File: drivers/counter/Kconfig.ambiq [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+
+config COUNTER_AMBIQ
+	bool "Ambiq Counter Driver"
+	default y
+	depends on DT_HAS_AMBIQ_COUNTER_ENABLED
+	select AMBIQ_HAL
+	select AMBIQ_HAL_USE_TIMER
+	help
+	  Enables the Counter driver for Ambiq drvices.
```
Comment (fkokosinski) [inline drivers/counter/Kconfig.ambiq:13]:
```suggestion
	  Enables the Counter driver for Ambiq devices.
```

**Review by fkokosinski — CHANGES_REQUESTED** [review]
Typo ;)

**Review by tgorochowik — COMMENTED** [review]
I think we shoud add a `_timer` suffix to the files to follow the pattern other drivers use + it will be easier to distinguish the drivers at a later point, when `_rtc` is added

File: drivers/counter/Kconfig.ambiq (hunk 1 above)
Comment (msierszulski) [reply drivers/counter/Kconfig.ambiq]: Thanks ;)

**Comment by msierszulski** [comment]
> I think we shoud add a `_timer` suffix to the files to follow the pattern other drivers use + it will be easier to distinguish the drivers at a later point, when `_rtc` is added

Thanks @tgorochowik! I have renamed the file.

**Review by fkokosinski — DISMISSED** [review]

**Review by tgorochowik — DISMISSED** [review]

**Review by fkokosinski — APPROVED** [review]

**Review by tgorochowik — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**